src/chain/chain.rs

```diff
@@ -408,35 +381,29 @@ impl Chain {
         &self.our_id
     }
 
-    /// Returns our own current section info, or `None`, if not available.
-    pub fn opt_our_info(&self) -> Option<&SectionInfo> {
-        self.our_infos.last().map(|&(ref si, _)| si)
-    }
-
     /// Returns our own current section info.
     pub fn our_info(&self) -> &SectionInfo {
-        // TODO: Replace `our_infos` with a new `NonemptyVec` type that statically guarantees that
-        // it's never empty.
-        &unwrap!(self.opt_our_info())
+        self.state.our_info()
     }
 
     /// Returns our own current section's prefix.
     pub fn our_prefix(&self) -> &Prefix<XorName> {
-        self.opt_our_info()
-            .map_or(&DEFAULT_PREFIX, SectionInfo::prefix)
+        self.state.our_prefix()
     }
 
-    /// Returns our current chain state.
-    pub fn state(&self) -> &ChainState {
+    /// Returns the shared state of the section.
+    pub fn shared_state(&self) -> &SharedState {
```

Comment:
Not sure why we made the full SharedState accessible while before we only allow ChainState.

Reply:
I can remove it, but I anticipate we are going to need it eventually (at least in tests).